### ISSUE TITLE
module version 4.23.1 causes ios 13 crash on app launch

### ISSUE THREAD

**Issue by ericpolkadoc** [adjust/titanium_sdk#26]
Hi,

We've updated to version 4.23.1, and on our ios 13 devices, our app now crashes on start up.  We're getting an error in terminal:

iOS 13 - Library not loaded: AppTrackingTransparency.framework/AppTrackingTransparency

Understandably this is an ios14 library, how is the module built to work with ios13?  I've read the section in the docs here, but there's nothing specifically noted.

https://github.com/adjust/titanium_sdk#att-framework

I've also read about Apple's delay of forcing the opt-in, and Adjust's suggested ways of handling the opt-in.  Do you have any working examples on how we would use the app tracking authorization handler?  Do we put this in our alloy.js and run our Adjust code in the ATTrackingManagerAuthorizationStatusAuthorized case?

https://github.com/adjust/titanium_sdk#app-tracking-authorisation-wrapper

Thanks!

**Comment by uerceg** [comment]
Hi @ericpolkadoc 

Thanks for reporting. Hm. Interesting. I think I have an idea what might be wrong. I'll try to generate updated `ti.adjust-*` archives for you with fix which might solve this and post them in here for you to check. Will keep you posted.

**Comment by uerceg** [comment]
Hi @ericpolkadoc 

Here's [ti.adjust-iphone-4.23.1.zip](https://github.com/adjust/titanium_sdk/files/5845366/ti.adjust-iphone-4.23.1.zip) build with potential fix for this issue you're seeing with `AppTrackingTransparency.framework` on iOS 13 devices. If you can give it a shot and let us know if it maybe fixed the issue, would be great.

In theory it should work with pre iOS 14 devices as well, but I think that the issue is that framework is linked directly instead of weakly and this is what fix build from above is trying to do, so curious to hear if it helped you with the issue.

Regarding ATT dialog question: I guess this is probably something which is better to be discussed with either your dedicated account manager or maybe with support@adjust.com since we (developers) are probably not the right place to give answers about ways how this might be implemented in your app. From pure technical perspective, I'd say you can chose to do any of two scenarios:

- The one you suggested, where you first wanna show the dialog and then only start Adjust SDK in case you get authorisation from your user.
- Or simply start SDK ASAP as suggested in README and fire dialog question in parallel with this flow. In this scenario, SDK will do the tracking and anything which gets sent to our end before your user provides an answer will (at this moment, until Apple still doesn't have this enforced) be reported as `not determined` status. After your user gives an answer to dialog, SDK will communicate this choice to our backend and we're gonna know that state has updated and act accordingly to that new state.

I guess which one of these two scenarios you should pick depends on how business logic in your app works and what do you expect to get from Adjust platform. But like said, in order to get better answers from that side, better to talk to our support team.

**Comment by ericpolkadoc** [comment]
Hi @uerceg ,

Thanks for getting back to me.  That seemed to do the trick, we're able to build ios13 now.  However reading that Apple are not ready to enforce the new opt-in, we're holding off on upgrading our modules.  But good to know we were able to help sort out the iOS13 issue with you for the future.

I do have a question for you.  In my scenario #1, we're beholden to the user's choice if they want to share their IDFA (Im guessing most will not) - will our deeplinking not work if the IDFA becomes all 0000's?  If the deeplinking will still work, am I correct in understanding that the reporting is going to be way off, as unique IDFA's get converted to multiple 0000's?  Just trying to understand if we're losing all functionality if someone opts out, or just the reporting part.

In your scenario #2, will the IDFA still used until the permission is acted on by the user?  What does a "not determined" status look like to our reporting?

thanks!

**Comment by uerceg** [comment]
Hi @ericpolkadoc 

We just released [v4.23.2](https://github.com/adjust/titanium_sdk/releases/tag/v4.23.2) update with fix from the test build I sent you few comments ago. Thank you one more time for spotting this. 👍 

Back to your questions from above. I'm gonna assume you're asking about deferred deep linking in here (because direct deep linking when your user has app already installed and clicks on link which opens it works just fine regardless of IDFA) and maybe fact that attaching device ID (IDFA) in tracker URL will probably start not to make much sense (like you said - probably significant number of people will click allow on that dialog, but then again that's also debatable, depends on how you phrase this, what's the purpose, etc etc, but let's assume that most probably lack of IDFA will probably be higher than LAT rates these days). Deferred deep linking will definitely work when the user consents. If the user does not consent it will depend on what Apple allows. Technically, we have various solutions that would be able to trigger the deferred deep link, however it depends on what solution Apple allows for a user who has not consented. SKAdNetwork, for example, does not support deferred deep linking.

In scenario 2 during this "transition period" (where Apple still doesn't enforce the policy) if IDFA is attempted to be read before ATT dialog is displayed then:
- It will return actual value if user has **Allow Apps to Request to Track** setting ON in `Settings -> Privacy` section on iOS device and **att_status** will be `not recognised`.
- It will return zeros if above mentioned setting is OFF. **att_status** will be `denied`.

The day when Apple enforces their rules, then attempt to read IDFA will probably result with zeros regardless of the toggle setting for above mentioned setting.

How various **att_status** values reflect in reports you're getting (in dashboard or elsewhere) - best if you check with our support team because they're gonna be able to help you with questions like that one for sure.

In case you still have any questions feel free to ping in here or reopen the issue if it still persists.

Cheers!
